src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,44 @@
-import React from 'react';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from '../../app/hooks';
+import { actions as filterActions } from '../../features/filter';
+
+import { StatusTodos } from '../../types/StatusTodos';
+import { ShowTodos } from '../../types/ShowTodos';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useDispatch();
+  const query = useAppSelector(state => state.filter.query);
+
+  const handleFilterStatus = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    switch (event.target.value) {
+      case StatusTodos.Active:
+        dispatch(filterActions.setStatus(ShowTodos.Active));
+        break;
+
+      case StatusTodos.Completed:
+        dispatch(filterActions.setStatus(ShowTodos.Completed));
+        break;
+
+      default:
+        dispatch(filterActions.setStatus(ShowTodos.All));
+        break;
+    }
+  };
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    dispatch(filterActions.setQuery(event.target.value));
+  };
+
+  const clearQuery = () => dispatch(filterActions.setQuery(null));
+
   return (
-    <form
-      className="field has-addons"
-      onSubmit={event => event.preventDefault()}
-    >
+    <form className="field has-addons">
```

Comment:
You should prevent the default form submission event to avoid page reload.

Reply:
Fix